src/main/java/org/mozilla/jss/ssl/SSLAlertEvent.java

```diff
@@ -57,11 +59,19 @@ public SSLAlertEvent(SSLFDProxy proxy, SSLAlertLevel level, SSLAlertDescription
     }
 
     public SSLSocket getSocket() {
-        return (SSLSocket) getSource();
+        Object obj = getSource();
+	if( obj != null && obj instanceof SSLSocket) {
+            return (SSLSocket) obj;
+	}
+	return null;
```

Comment:
This (and other instances like it) could be ternary operators if one was so inclined. Also you don't need to check for null, `instanceof` does this implicitly:

```
    public SSLSocket getSocket() {
        Object obj = getSource();
        return obj instanceof SSLSocket ? (SSLSocket) obj : null;
    }
```